### PR TITLE
BUG: Fix issue in preserving index name on empty DataFrame

### LINKED ISSUE #36527: BUG: Different behaviour for index after assign dataframe new column by series
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of pandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of pandas.

---

#### Code Sample, a copy-pastable example

With pandas version 1.1.1

```python
import pandas as pd

df = pd.DataFrame()
series = pd.Series(1.23, index=pd.RangeIndex(4, name="series_index"))
df['series'] = series
print(df)
```

#### Problem description

With pandas version 1.1.1, output:
```
                         series
series_index        
0                       1.23
1                       1.23
2                       1.23
3                       1.23
```
We perceive index from series

With pandas version 1.1.2, output:
```
   series
0    1.23
1    1.23
2    1.23
3    1.23
```
We lost the index from series

#### Expected Output
I am not sure about it. Could you explain is bug or feature of the latest version?
We used to perceive index from series and it worked before.

#### Output of ``pd.show_versions()``

<details>

[INSTALLED VERSIONS
------------------
commit           : 2a7d3326dee660824a8433ffd01065f8ac37f7d6
python           : 3.8.5.final.0
python-bits      : 64
OS               : Linux
OS-release       : 4.19.104-microsoft-standard
Version          : #1 SMP Wed Feb 19 06:37:35 UTC 2020
machine          : x86_64
processor        : 
byteorder        : little
LC_ALL           : None
LANG             : None
LOCALE           : en_US.UTF-8
pandas           : 1.1.2
numpy            : 1.19.2
pytz             : 2020.1
dateutil         : 2.8.1
pip              : 20.2.3
setuptools       : 50.3.0
Cython           : None
pytest           : 6.0.2
hypothesis       : None
sphinx           : None
blosc            : None
feather          : None
xlsxwriter       : None
lxml.etree       : None
html5lib         : None
pymysql          : None
psycopg2         : 2.8.6 (dt dec pq3 ext lo64)
jinja2           : 2.10.3
IPython          : None
pandas_datareader: None
bs4              : None
bottleneck       : None
fsspec           : None
fastparquet      : None
gcsfs            : None
matplotlib       : None
numexpr          : None
odfpy            : None
openpyxl         : None
pandas_gbq       : None
pyarrow          : None
pytables         : None
pyxlsb           : None
s3fs             : None
scipy            : None
sqlalchemy       : 1.3.19
tables           : None
tabulate         : 0.8.7
xarray           : None
xlrd             : 1.2.0
xlwt             : None
numba            : None
]

</details>

### PR DESCRIPTION
**Pull request by Dr-Irv**
- [x] closes #36527
- [x] tests added / passed
   - `tests/indexing/test_partial.py:tesst_index_name_empty`
- [x] passes `black pandas`
- [x] passes `git diff upstream/master -u -- "*.py" | flake8 --diff`
- [x] whatsnew entry
   - v1.1.3

### CODE REVIEW

**Review by WillAyd — APPROVED** [review]
lgtm @jorisvandenbossche

**Comment by jreback** [comment]
thanks @Dr-Irv

**Comment by jorisvandenbossche** [comment]
@MeeseeksDev backport to 1.1.x

File: pandas/tests/indexing/test_partial.py [hunk 1]
```diff
@@ -672,3 +672,11 @@ def test_index_name_empty(self):
...
 
         tm.assert_frame_equal(df, expected)
+
+        # GH 36527
+        df = pd.DataFrame()
+        series = pd.Series(1.23, index=pd.RangeIndex(4, name="series_index"))
+        df["series"] = series
+        expected = pd.DataFrame(
+            {"series": [1.23] * 4}, index=pd.RangeIndex(4, name="series_index")
+        )
```
Comment (jorisvandenbossche) [inline pandas/tests/indexing/test_partial.py:682]: I think you forgor the assert here ;)

(thanks for the PR for the rest!)

Comment (jorisvandenbossche) [reply pandas/tests/indexing/test_partial.py:682]: -> https://github.com/pandas-dev/pandas/pull/36546

**Pull request merged**